### PR TITLE
Generate new Provisioning Client for the Automation Framework

### PR DESCRIPTION
**Pull request by igomez06**
**Framework Updates:**

- Moved the the Kube API provisiong package outside of rancher, but kept it in the clients package. This way integration tests can use the provisioning Kube API provisioning client if needed.

- Updated the generated management clients. Updated the hosted provider extensions to account for the new management client updates.
 
- Implemented a new schema for the provisioing.cattle.io API, and also updated the codegen package to generate the provisioing.cattle.io client, as well as the management.cattle.io clients.

- Updated the rancher client to account for the new provisioning client. Also added helper to reinstantiate a client for when permissions change for a user. Added a helper that instantiates a kube api provisioning client.

**Validation  Test Updates:**
- Updated tests and packages to account for the new provisioning.cattle.io client.

- Fixed Custom cluster tests.

### CODE REVIEW

**Review by caliskanugur — APPROVED** [review]
Looks good to me

File: tests/framework/clients/k3d/k3d.go [hunk 1]
```diff
@@ -89,8 +90,12 @@ func CreateAndImportK3DCluster(client *rancher.Client, name string) (*apisV1.Clu
 		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to create k3d cluster")
 	}
 
+	kubeProvisioningClient, err := client.GetKubeAPIProvisioningClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to instaniated kube api provisioning client")
```
Comment (slickwarren) [inline tests/framework/clients/k3d/k3d.go:95]: typo on instantiate?

**Review by slickwarren — COMMENTED** [review]
small typo noticed, otherwise approved once test freeform job posted.

File: tests/framework/clients/k3d/k3d.go (hunk 1 above)
Comment (igomez06) [reply tests/framework/clients/k3d/k3d.go]: Fixed, and I sent you the freeform job as well Caleb

**Review by caliskanugur — DISMISSED** [review]
Looks good to me

**Review by slickwarren — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]
rebase approved

**Review by caliskanugur — APPROVED** [review]

**Pull request merged**